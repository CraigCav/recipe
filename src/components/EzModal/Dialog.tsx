import React, {useRef, useContext, useEffect} from 'react';
import {useDialog, DialogProps} from 'reakit/Dialog';
import {Portal} from 'reakit/Portal';
import {PortalContext} from '../EzPortal';

const isSSR = typeof document === 'undefined';

export const Dialog: React.FC<DialogProps> = ({children, ...props}) => {
  const {current: hostNode} = useContext(PortalContext);
  const initialFocusRef = useRef<HTMLElement>();
  const ref = useRef<HTMLElement>();
  const isOpen = props.visible;

  useEffect(() => {
    if (isOpen) initialFocusRef.current = hostNode?.ownerDocument?.activeElement as HTMLElement;
  }, [hostNode, isOpen]);

  const preventBodyScroll = !isSSR && hostNode?.ownerDocument === document;
  const initialFocus = 'unstable_finalFocusRef';
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  const {wrapElement, ...dialog} = useDialog(
    {[initialFocus]: initialFocusRef, preventBodyScroll, ...props},
    {ref}
  );

  return (
    // eslint-disable-next-line no-underscore-dangle
    <div {...dialog} className={`${Portal.__className} ${props.className}`}>
      <PortalContext.Provider value={ref}>{props.visible && children}</PortalContext.Provider>
    </div>
  );
};
