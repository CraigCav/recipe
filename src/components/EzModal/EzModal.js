import React from 'react';
import PropTypes from 'prop-types';
import ReactModal from 'react-modal';
import EzButton from '../EzButton';
import EzHeading from '../EzHeading';
import EzLayout from '../EzLayout';
import {
  CloseButton,
  ButtonFooter,
  HeaderContainer,
  ContentContainer,
  ModalContainer,
  reactModalFromTheme,
  reactModalHtmlOpen,
  reactModalAfterOpen,
  reactModalOverlay,
} from './EzModal.styles';
import {standard} from '../../themes';

const CloseIcon = ({theme, onClick, dismissLabel}) => (
  <CloseButton role="button" onClick={onClick} theme={theme}>
    <svg
      aria-label={dismissLabel}
      width="16px"
      height="16px"
      viewBox="0 0 16 16"
      version="1.1"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g id="Artboard" stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
        <path
          d="M6.5,6.5 L6.5,-1 L9.5,-1 L9.5,6.5 L17,6.5 L17,9.5 L9.5,9.5 L9.5,17 L6.5,17 L6.5,9.5 L-1,9.5 L-1,6.5 L6.5,6.5 Z"
          id="Close-Button"
          fill="#3A81BE"
          transform="translate(8.000000, 8.000000) rotate(-315.000000) translate(-8.000000, -8.000000) "
        />
      </g>
    </svg>
  </CloseButton>
);

class EzModal extends React.Component {
  componentDidMount() {
    if (this.props.appElement) {
      ReactModal.setAppElement(this.props.appElement);
    }
  }

  render() {
    const {
      children,
      headerText,
      destructive,
      dismissLabel,
      isOpen,
      isSubmitting,
      onDismiss,
      onSubmit,
      submitLabel,
      theme,
    } = this.props;

    return (
      <ReactModal
        isOpen={isOpen}
        onRequestClose={onDismiss}
        className={reactModalFromTheme({theme})}
        overlayClassName={reactModalOverlay}
        afterOpenClassName={reactModalAfterOpen}
        htmlOpenClassName={reactModalHtmlOpen}
      >
        {isOpen && (
          <ModalContainer theme={theme}>
            <HeaderContainer theme={theme}>
              <EzHeading size="2">{headerText}</EzHeading>
              <CloseIcon theme={theme} dismissLabel={dismissLabel} onClick={onDismiss} />
            </HeaderContainer>

            <ContentContainer theme={theme}>{children}</ContentContainer>

            <ButtonFooter theme={theme}>
              <EzLayout layout={{base: 'stack', medium: 'basic'}}>
                {submitLabel && (
                  <EzButton
                    use="primary"
                    destructive={destructive}
                    onClick={onSubmit}
                    loading={isSubmitting}
                    theme={theme}
                  >
                    {submitLabel}
                  </EzButton>
                )}
                <EzButton use="secondary" disabled={isSubmitting} onClick={onDismiss} theme={theme}>
                  {dismissLabel}
                </EzButton>
              </EzLayout>
            </ButtonFooter>
          </ModalContainer>
        )}
      </ReactModal>
    );
  }
}

EzModal.propTypes = {
  /**
   * A query selector identifying the root of your app.
   * Ezmodal uses this selector to indicate to screen readers that this content
   * should be hidden (via the aria-hidden attribute) while the modal is open.
   */
  appElement: PropTypes.string,

  /**
   * Arbitrary children for the modal contents
   */
  children: PropTypes.node.isRequired,

  /**
   * Text to display in the header
   */
  headerText: PropTypes.string.isRequired,

  /**
   * When true will display the modal in destructive styles (aka a destructive submit button)
   */
  destructive: PropTypes.bool,

  /**
   * Label for the dismiss button. Omitting this prop will suppress the dismiss button
   */
  dismissLabel: PropTypes.string.isRequired,

  /**
   * Dictates whether the modal is open. You must still render the component even
   * if isOpen is false to allow for the opening transition to occur (if present)
   */
  isOpen: PropTypes.bool.isRequired,

  /**
   * When true will render the modal with appropriate styles to signify a loading state
   * Useful when the submit handler has an asynchronous success condition
   */
  isSubmitting: PropTypes.bool,

  /**
   * Callback for when dismiss button is clicked.
   */
  onDismiss: PropTypes.func,

  /**
   * Callback for when submit button is clicked.
   */
  onSubmit: PropTypes.func,

  /**
   * Label for the submit button. Omitting this prop will suppress the submit button
   */
  submitLabel: PropTypes.string,

  /**
   * Theme object to control the style of the modal
   */
  theme: PropTypes.object,
};

EzModal.defaultProps = {
  appElement: '#root',
  destructive: false,
  isSubmitting: false,
  onDismiss() {},
  onSubmit() {},
  submitLabel: null,
  theme: standard,
};

export default EzModal;
