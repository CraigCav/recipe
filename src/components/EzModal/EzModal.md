---
name: Modal
title: Modal
path: "/components/ez-modal"
---

Modals display content in a layer on top of the application. Modals should be used when it’s necessary to interrupt the user or ensure their attention is focused on a specific chunk of information. Modals can also be used when you’d like the user to accomplish a lower-priority action without changing the state of the page they’re working in.

---

## Best Practices
Modals should:
   * Provide the user with clear paths forward (a primary action) and out of the modal (close icon in the upper right). For confirmation modals, include a secondary “no” or “cancel” action.
   * Use a footer to provide a consistent location for key actions. For modals with many equally-weighted actions, a footer isn’t necessary and the actions can live in the body of the modal.

Modals should not:
   * Create a “wizard” flow with three or more steps. A secondary step with a back button is acceptable, but discouraged.
   * Don’t use a “cancel” or “close” button on informational modals or editing modals.

## Examples
### Informational
Informational modals are used to focus the user’s attention on a specific chunk of information. Use when providing information about something that’s not essential to completing the actions on the page (e.g. more information about ezDispatch), especially if content formatting is needed or there’s too much content to display in a tooltip.

### Editing
Editing modals can contain any form elements and should be kept as brief as possible. Long or complex forms should almost always be done in a new section/page instead of a modal.

### Confirmation
Use a confirmation modal to ask the user to confirm changes they’re about to make or confirm a destructive action they’re about to take.

#### Make sure to:
   * Use destructive buttons to clearly designate actions with destructive consequences
   * Provide a cancel option with a clear label. Make sure the copy isn’t confusing when placed next to a destructive action (cancel order, cancel).
   * Provide appropriate detail so the user can make an informed decision.

## Examples

Note all examples include a div with class appElement. In practice this div
will be your application's root element and does not need to reside directly
with your modal

### Basic Modal

```jsx live
<Component initialState={{isOpen: false}}>
  {({state, setState}) => (
    <React.Fragment>
      <div className="appElement">
        <button onClick={() => setState({isOpen: true})}>Button</button>
        <EzModal
          isOpen={state.isOpen}
          appElement=".appElement"
          onSubmit={() => {
            alert('Submitted!');
            setState({isOpen: false});
          }}
          submitLabel="Submit"
          onDismiss={() => {
            alert('Dismissed!');
            setState({isOpen: false});
          }}
          dismissLabel="Dismiss"
          headerText="Header goes here"
        >
          Modal content goes here!
        </EzModal>
      </div>
    </React.Fragment>
  )}
</Component>
```

### Only dismiss button

For a purely informational modal you can omit the submitLabel prop which will suppress the submit button

```jsx live
<Component initialState={{isOpen: false}}>
  {({state, setState}) => (
    <React.Fragment>
      <div className="appElement">
        <button onClick={() => setState({isOpen: true})}>Button</button>
        <EzModal
          isOpen={state.isOpen}
          appElement=".appElement"
          onDismiss={() => setState({isOpen: false})}
          dismissLabel="Dismiss"
          headerText="Header goes here"
        >
          Modal content goes here!
        </EzModal>
      </div>
    </React.Fragment>
  )}
</Component>
```

### Destructive

If the submit action is destructive you can set the destructive prop

```jsx live
<Component initialState={{isOpen: false}}>
  {({state, setState}) => (
    <React.Fragment>
      <div className="appElement">
        <button onClick={() => setState({isOpen: true})}>Button</button>
        <EzModal
          destructive
          isOpen={state.isOpen}
          appElement=".appElement"
          onSubmit={() => setState({isOpen: false})}
          submitLabel="Submit"
          onDismiss={() => setState({isOpen: false})}
          dismissLabel="Dismiss"
          headerText="Header goes here"
        >
          Modal content goes here!
        </EzModal>
      </div>
    </React.Fragment>
  )}
</Component>
```

### isSubmitting

You can set the isSubmitting to trigger the loading state for the modal

```jsx live
<Component initialState={{isOpen: false}}>
  {({state, setState}) => (
    <React.Fragment>
      <div className="appElement">
        <button onClick={() => setState({isOpen: true})}>Button</button>
        <EzModal
          isSubmitting
          isOpen={state.isOpen}
          appElement=".appElement"
          onSubmit={() => setState({isOpen: false})}
          submitLabel="Submit"
          onDismiss={() => setState({isOpen: false})}
          dismissLabel="Dismiss"
          headerText="Header goes here"
        >
          Modal content goes here!
        </EzModal>
      </div>
    </React.Fragment>
  )}
</Component>
```

---

## Related components

\*
