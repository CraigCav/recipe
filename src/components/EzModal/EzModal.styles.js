import styled, {css} from 'react-emotion';
import {padding, position} from 'polished';
import ReactModal from 'react-modal';

const circleHeight = '40px';
const closeIconBorderRadius = '50%';
const closeIconHeight = '16px';

export const CloseButton = styled.button`
  background-color: transparent;
  border: 0;
  border-radius: ${closeIconBorderRadius};
  cursor: pointer;
  margin-right: calc((${closeIconHeight} - ${circleHeight}) / 2);
  margin-top: calc((${closeIconHeight} - ${circleHeight}) / 2);
  outline: none;
  padding: calc((${circleHeight} - ${closeIconHeight}) / 2);

  :hover {
    background-color: ${props => props.theme.colors.grays[200]};
  }
`;

export const Modal = styled(ReactModal)`
  left: 50%;
  min-height: 100%;
  min-width: 100%;
  overflow-y: hidden;
  position: fixed;
  top: 150%;
  transform: translate(-50%, -50%);
  transition: top 0.5s;

  &.ReactModal__Content--after-open {
    top: 50%;
  }

  @media screen and (min-width: ${({theme}) => theme.breakpoints.medium}) {
    border-radius: 12px;
    left: 50%;
    min-height: 0;
    min-width: 0;
    overflow: hidden;
    top: 50%;
    width: 500px;
  }

  :focus {
    outline: 0;
  }
`;

export const reactModalOverlay = css`
  &.ReactModal__Overlay--after-open {
    background-color: rgba(0, 0, 0, 0.6);
    ${position('fixed', 0, 0, 0, 0)};
  }
`;

export const reactModalHtmlOpen = css`
  overflow: hidden;
`;

export const reactModalAfterOpen = css`
  top: 50%;
`;

export const ModalContainer = styled.div`
  background: ${props => props.theme.colors.white};
  display: flex;
  flex-direction: column;
  height: 100%;
  max-height: 100vh;
  position: absolute;
  width: 100%;

  @media screen and (min-width: ${props => props.theme.breakpoints.medium}) {
    max-height: calc(100vh - ${props => props.theme.spacing.xl4});
    position: relative;
  }
`;

export const HeaderContainer = styled.div`
  border-bottom: 1px solid ${props => props.theme.colors.grays[300]};
  display: flex;
  flex: none;
  justify-content: space-between;
  padding: ${props => props.theme.spacing.md} ${props => props.theme.spacing.xl};

  @media screen and (min-width: ${props => props.theme.breakpoints.medium}) {
    border-bottom: none;
    ${({
      theme: {
        spacing: {xl, xl2},
      },
    }) => padding(xl2, xl2, xl, xl2)};
  }
`;

export const ContentContainer = styled.div`
  flex: auto;
  overflow-y: auto;
  ${({
    theme: {
      spacing: {xl},
    },
  }) => padding(xl, xl, xl, xl)};

  @media screen and (min-width: ${props => props.theme.breakpoints.medium}) {
    ${({
      theme: {
        spacing: {xl2},
      },
    }) => padding(0, xl2, xl2, xl2)};
  }
`;

export const ButtonFooter = styled.div`
  background: ${props => props.theme.colors.grays[200]};
  border-top: 1px solid ${props => props.theme.colors.grays[300]};
  flex: none;
  padding: ${props => props.theme.spacing.xl2};
`;
