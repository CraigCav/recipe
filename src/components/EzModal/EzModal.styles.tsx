import {css} from 'react-emotion';
import {DialogContent, DialogOverlay} from '@reach/dialog';
import styled from '../../themes/styled';

const closeIconBorderRadius = '50%';
const clickTargetPadding = '12px';

export const CloseButton = styled.button`
  background-color: transparent;
  border: 0;
  border-radius: ${closeIconBorderRadius};
  cursor: pointer;
  outline: none;
  margin: -${clickTargetPadding}; /* negate the impact of the extra padding on the elements containers height */
  padding: ${clickTargetPadding};

  :hover {
    background-color: ${props => props.theme.colors.interactive.hover.background};
  }
  :active {
    background-color: ${props => props.theme.colors.interactive.active.background};
  }
`;

export const Overlay = styled(DialogOverlay)`
  align-items: center;
  background-color: rgba(0, 0, 0, 0.6);
  display: flex;
  flex-direction: column;
  justify-content: center;
  overflow: hidden;
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
`;

export const ModalContainer = styled(DialogContent)`
  background: ${props => props.theme.colors.white};
  display: flex;
  flex-direction: column;
  height: 100%;
  max-height: 100vh;
  outline: none;
  width: 100%;

  @media screen and (min-width: ${props => props.theme.breakpoints.medium}) {
    border-radius: 12px;
    height: auto;
    overflow: hidden; /* so the border-radius is applies to child content  */
    max-height: calc(100vh - ${props => props.theme.spacing.xl4});
    width: 500px;
  }
`;

const containerPadding = ({theme: {spacing, breakpoints}}) => css`
  padding: ${spacing.md} ${spacing.xl};

  @media screen and (min-width: ${breakpoints.medium}) {
    padding: ${spacing.xl} ${spacing.xl2};
  }
`;

export const HeaderContainer = styled.div`
  ${props => containerPadding({theme: props.theme})}
  align-items: baseline;
  border-bottom: 1px solid ${props => props.theme.colors.border.subtle};
  display: flex;
  flex: none;
  justify-content: space-between;

  @media screen and (min-width: ${props => props.theme.breakpoints.medium}) {
    border-bottom: none;
  }
`;

export const ContentContainer = styled.div`
  ${props => containerPadding({theme: props.theme})}
  flex: auto;
  overflow-y: auto;

  @media screen and (min-width: ${props => props.theme.breakpoints.medium}) {
    padding-top: 0;
  }
`;

export const ButtonFooter = styled.div`
  ${props => containerPadding({theme: props.theme})}
  background: ${props => props.theme.colors.page.background};
  border-top: 1px solid ${props => props.theme.colors.border.subtle};
  flex: none;
`;
