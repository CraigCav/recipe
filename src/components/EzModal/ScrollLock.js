import React from 'react';
import {css} from 'react-emotion';

export const preventScroll = css`
  overflow: hidden;
`;

const disableScroll = () => document && document.querySelector('html').classList.add(preventScroll);
const enableScroll = () =>
  document && document.querySelector('html').classList.remove(preventScroll);

export default class ScrollLock extends React.Component {
  componentDidMount() {
    disableScroll();
  }

  componentWillUnmount() {
    enableScroll();
  }

  render() {
    return null;
  }
}
