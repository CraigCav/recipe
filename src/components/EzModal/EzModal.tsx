/** @jsx jsx */
import {jsx} from '@emotion/core';
import React from 'react';
import EzButton from '../EzButton';
import EzHeading from '../EzHeading';
import EzLayout from '../EzLayout';
import {
  ButtonFooter,
  HeaderContainer,
  ContentContainer,
  StyledOverlay,
  dialogStyles,
} from './EzModal.styles';
import FocusScope from '../FocusScope';
import CloseButton from '../CloseButton';
import {useUniqueId} from '../../utils/hooks';
import EzPortal from '../EzPortal';
import {ScrollLock} from './ScrollLock';
import {RequireAtLeastOne} from '../../typings/utility';

type Props = {
  headerText: string;
  destructive?: boolean;
  isOpen: boolean;
  isSubmitting?: boolean;
  onDismiss?: () => void;
  onSubmit?: () => void;
  appElement?: string;
  submitLabel?: string;
  dismissLabel?: string;
};

type PropsWithRequiredLabels = Props & RequireAtLeastOne<'submitLabel' | 'dismissLabel'>;

const useDialogBackdrop = ({onDismiss}) => ({
  onClick: e => e.target === e.currentTarget && onDismiss(),
});

const useDialog = ({onDismiss}) => ({
  role: 'dialog',
  'aria-modal': true,
  tabIndex: -1,
  onKeyDown: (event: React.KeyboardEvent) => {
    if (event.defaultPrevented) return;
    if (event.key !== 'Escape') return;
    event.stopPropagation();
    onDismiss();
  },
});

const EzModal: React.FC<PropsWithRequiredLabels> = ({
  children,
  headerText,
  destructive,
  dismissLabel,
  isOpen,
  isSubmitting,
  onDismiss = () => {},
  onSubmit,
  submitLabel,
}) => {
  const labelId = useUniqueId();
  const backdrop = useDialogBackdrop({onDismiss});
  const dialog = useDialog({onDismiss});

  if (!isOpen) return null;

  return (
    <EzPortal>
      <StyledOverlay {...backdrop}>
        <ScrollLock />
        <FocusScope contain restoreFocus autoFocus>
          <div {...dialog} css={dialogStyles()} aria-labelledby={labelId}>
            <HeaderContainer>
              <EzHeading size="2" id={labelId}>
                {headerText}
              </EzHeading>
              {dismissLabel && (
                <CloseButton
                  css={{alignSelf: 'center'}}
                  tabIndex={-1}
                  label={dismissLabel}
                  onClick={onDismiss}
                />
              )}
            </HeaderContainer>

            <ContentContainer>{children}</ContentContainer>

            <ButtonFooter>
              <EzLayout layout={{base: 'stack', medium: 'basic'}}>
                {submitLabel && (
                  <EzButton
                    use="primary"
                    destructive={destructive}
                    onClick={onSubmit}
                    loading={isSubmitting}
                  >
                    {submitLabel}
                  </EzButton>
                )}
                {dismissLabel && (
                  <EzButton use="secondary" disabled={isSubmitting} onClick={onDismiss}>
                    {dismissLabel}
                  </EzButton>
                )}
              </EzLayout>
            </ButtonFooter>
          </div>
        </FocusScope>
      </StyledOverlay>
    </EzPortal>
  );
};

export default EzModal;
