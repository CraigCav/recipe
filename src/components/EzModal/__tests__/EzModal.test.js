import React from 'react';
import {fireEvent, cleanup} from 'react-testing-library';
import 'jest-dom/extend-expect';
import EzModal from '../EzModal';
import {standard} from '../../../themes';

afterEach(cleanup);

describe('EzModal', () => {
  beforeEach(() => {
    // suppress import style warning from Reach Modal
    jest.spyOn(CSSStyleDeclaration.prototype, 'getPropertyValue').mockReturnValue('1');
  });

  it('should render with default styles', () => {
    const {baseElement} = fullRender(
      <EzModal isOpen submitLabel="Submit" dismissLabel="Dismiss" headerText="Header">
        children
      </EzModal>
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('does not render the children if not open', () => {
    const {queryByText} = fullRender(
      <EzModal isOpen={false} dismissLabel="dismiss" headerText="header">
        <div>foo</div>
      </EzModal>
    );
    expect(queryByText('foo')).toBeNull();
  });

  it('calls submit handler when submit button is clicked', () => {
    const clickSpy = jest.fn();
    const submitLabel = 'submit';
    const {getByText} = fullRender(
      <EzModal
        isOpen
        submitLabel={submitLabel}
        onSubmit={clickSpy}
        dismissLabel="dismiss"
        headerText="header"
      >
        Test
      </EzModal>
    );

    fireEvent.click(getByText(submitLabel));
    expect(clickSpy).toHaveBeenCalled();
  });

  it('calls dismiss handler when dismiss button is clicked', () => {
    const clickSpy = jest.fn();
    const dismissLabel = 'dismiss';
    const {getByText} = fullRender(
      <EzModal
        isOpen
        dismissLabel={dismissLabel}
        submitLabel="submit"
        onDismiss={clickSpy}
        headerText="header"
      >
        Test
      </EzModal>
    );

    fireEvent.click(getByText(dismissLabel));
    expect(clickSpy).toHaveBeenCalled();
  });

  it('passes through the destructive prop to the submit button ', () => {
    const submitLabel = 'submit';
    const {getByText} = fullRender(
      <EzModal
        isOpen
        submitLabel={submitLabel}
        destructive
        dismissLabel="dismiss"
        headerText="header"
      >
        test
      </EzModal>
    );

    const submitButton = getByText(submitLabel);
    expect(submitButton).toHaveStyle(`background-color: ${standard.colors.destructive}`);
  });

  it('disables the submit button when the form is submitting', () => {
    const submitLabel = 'submit';
    const {getByText} = fullRender(
      <EzModal
        isOpen
        isSubmitting
        submitLabel={submitLabel}
        dismissLabel="dismiss"
        headerText="header"
      >
        test
      </EzModal>
    );
    const submitButton = getByText(submitLabel);
    expect(submitButton).toHaveAttribute('disabled');
  });

  it('sets the disabled prop on dismiss button when isSubmitting is true', () => {
    const dismissLabel = 'dismiss';
    const {getByText} = fullRender(
      <EzModal isOpen isSubmitting dismissLabel={dismissLabel} headerText="header">
        test
      </EzModal>
    );
    const dismissButton = getByText(dismissLabel);
    expect(dismissButton).toHaveAttribute('disabled');
  });

  /**
   * Accessibility tests.
   */
  describe('Accessibility tests', () => {
    it('should meet accessibility guidelines when required labels / text are given', async () => {
      const {baseElement, unmount} = fullRender(
        <EzModal isOpen dismissLabel="dismiss" headerText="header">
          test
        </EzModal>
      );
      unmount();
      const actual = await axe(baseElement.innerHTML);
      expect(actual).toHaveNoViolations();
    });

    it('should meet accessibility guidelines when all labels / text are given', async () => {
      const {baseElement, unmount} = fullRender(
        <EzModal isOpen headerText="Header" submitLabel="submit" dismissLabel="dismiss">
          test
        </EzModal>
      );
      unmount();
      const actual = await axe(baseElement.innerHTML);
      expect(actual).toHaveNoViolations();
    });
  });
});
