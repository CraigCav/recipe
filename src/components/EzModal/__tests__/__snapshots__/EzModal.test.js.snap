// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EzModal should render with default styles 1`] = `
.recipe-21.ReactModal__Overlay--after-open {
  background-color: rgba(0,0,0,0.6);
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
}

.recipe-19 {
  left: 50%;
  min-height: 100%;
  min-width: 100%;
  overflow-y: hidden;
  position: fixed;
  top: 150%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  -webkit-transition: top 0.5s;
  transition: top 0.5s;
}

.recipe-19.ReactModal__Content--after-open {
  top: 50%;
}

@media screen and (min-width:48em) {
  .recipe-19 {
    border-radius: 12px;
    left: 50%;
    min-height: 0;
    min-width: 0;
    overflow: hidden;
    top: 50%;
    width: 500px;
  }
}

.recipe-19:focus {
  outline: 0;
}

.recipe-17 {
  background: #ffffff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  max-height: 100vh;
  position: absolute;
  width: 100%;
}

@media screen and (min-width:48em) {
  .recipe-17 {
    max-height: calc(100vh - 4.285714285714286rem);
    position: relative;
  }
}

.recipe-5 {
  border-bottom: 1px solid #e6e9eb;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 1.1428571428571428rem 1.7142857142857142rem;
}

@media screen and (min-width:48em) {
  .recipe-5 {
    border-bottom: none;
    padding-top: 2.2857142857142856rem;
    padding-right: 2.2857142857142856rem;
    padding-bottom: 1.7142857142857142rem;
    padding-left: 2.2857142857142856rem;
  }
}

.recipe-1 {
  font-size: 1.5rem;
  font-weight: 400;
  margin: 0;
}

.recipe-3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 0;
  background-color: transparent;
  border-radius: 50%;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: calc((16px - 40px) / 2);
  margin-top: calc((16px - 40px) / 2);
  outline: none;
  height: 40px;
  width: 40px;
}

.recipe-3:hover {
  background-color: #f4f7f8;
}

.recipe-7 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-top: 0;
  overflow-y: auto;
  padding-top: 1.7142857142857142rem;
  padding-right: 1.7142857142857142rem;
  padding-bottom: 0;
  padding-left: 1.7142857142857142rem;
}

@media screen and (min-width:48em) {
  .recipe-7 {
    padding-top: 0;
    padding-right: 2.2857142857142856rem;
    padding-bottom: 2.2857142857142856rem;
    padding-left: 2.2857142857142856rem;
  }
}

.recipe-15 {
  background: #f4f7f8;
  border-top: 1px solid #e6e9eb;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  padding: 1.7142857142857142rem;
}

.recipe-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

@media screen and (min-width:48em) {
  .recipe-13 {
    -webkit-align-items: normal;
    -webkit-box-align: normal;
    -ms-flex-align: normal;
    align-items: normal;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: normal;
    -webkit-justify-content: normal;
    -ms-flex-pack: normal;
    justify-content: normal;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }

  .recipe-13 > * {
    -webkit-flex-basis: auto;
    -ms-flex-preferred-size: auto;
    flex-basis: auto;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
  }
}

.recipe-13 > *:not(:first-child) {
  margin-top: 0.8571428571428571rem;
}

@media screen and (min-width:48em) {
  .recipe-13 > *:not(:first-child) {
    margin-top: 0;
    margin-left: 0;
  }

  .recipe-13 > *:not(:first-child) {
    margin-left: 0.8571428571428571rem;
  }
}

.recipe-9 {
  cursor: pointer;
  display: inline-block;
  outline: 0;
  font-family: Lato,'Helvetica Neue',Arial,Helvetica,sans-serif;
  font-weight: 700;
  line-height: 1em;
  min-height: 1em;
  margin-right: 0.2857142857142857rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  -webkit-tap-highlight-color: transparent;
  color: #ffffff;
  border-color: #3e90d6;
  border-color: #3e90d6;
  background-color: #3e90d6;
  border-radius: 4px;
  border-style: solid;
  border-width: 1px;
  box-shadow: 0 1px 1px 0 rgba(0,0,0,0.12);
  padding: 0.5714285714285714rem 1.1428571428571428rem;
}

.recipe-9:hover:enabled {
  border-color: #347ab5;
}

.recipe-9:focus:enabled {
  border-color: #2e6ca0;
}

.recipe-9:active:enabled {
  border-color: #2e6ca0;
}

.recipe-9:hover:enabled {
  background-color: #347ab5;
}

.recipe-9:active:enabled {
  background-color: #2e6ca0;
}

.recipe-11 {
  cursor: pointer;
  display: inline-block;
  outline: 0;
  font-family: Lato,'Helvetica Neue',Arial,Helvetica,sans-serif;
  font-weight: 700;
  line-height: 1em;
  min-height: 1em;
  margin-right: 0.2857142857142857rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  -webkit-tap-highlight-color: transparent;
  background-color: white;
  border: solid 1px;
  border-color: #ced4d9;
  color: #3e90d6;
  border-radius: 4px;
  border-style: solid;
  border-width: 1px;
  box-shadow: 0 1px 1px 0 rgba(0,0,0,0.12);
  padding: 0.5714285714285714rem 1.1428571428571428rem;
}

.recipe-11:hover:enabled {
  border-color: #afb4b8;
}

.recipe-11:focus:enabled {
  border-color: #3e90d6;
}

.recipe-11:active:enabled {
  border-color: #9a9fa2;
}

.recipe-11:hover:enabled {
  color: #347ab5;
}

.recipe-11:active:enabled {
  color: #2e6ca0;
}

<div
  aria-modal="true"
  className="ReactModal__Overlay ReactModal__Overlay--after-open recipe-21"
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  style={Object {}}
>
  <div
    className="ReactModal__Content ReactModal__Content--after-open recipe-19 recipe-20"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    style={Object {}}
    tabIndex="-1"
  >
    <div
      className="recipe-17 recipe-18"
    >
      <div
        className="recipe-5 recipe-6"
      >
        <h2
          className="recipe-0 recipe-1 recipe-2"
          size="2"
        >
          Header
        </h2>
        <button
          className="recipe-3 recipe-4"
          onClick={[Function]}
          role="button"
        >
          <svg
            aria-label="Dismiss"
            height="16px"
            version="1.1"
            viewBox="0 0 16 16"
            width="16px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fillRule="evenodd"
              id="Artboard"
              stroke="none"
              strokeWidth="1"
            >
              <path
                d="M6.5,6.5 L6.5,-1 L9.5,-1 L9.5,6.5 L17,6.5 L17,9.5 L9.5,9.5 L9.5,17 L6.5,17 L6.5,9.5 L-1,9.5 L-1,6.5 L6.5,6.5 Z"
                fill="#3A81BE"
                id="Close-Button"
                transform="translate(8.000000, 8.000000) rotate(-315.000000) translate(-8.000000, -8.000000) "
              />
            </g>
          </svg>
        </button>
      </div>
      <div
        className="recipe-7 recipe-8"
      >
        children
      </div>
      <div
        className="recipe-15 recipe-16"
      >
        <div
          className="recipe-13 recipe-14"
        >
          <button
            className="recipe-9 recipe-10"
            disabled={false}
            onClick={[Function]}
          >
            Submit
          </button>
          <button
            className="recipe-11 recipe-10"
            onClick={[Function]}
          >
            Dismiss
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
